types/qaengine/commonqa/commonqa.go

```diff
@@ -124,3 +125,29 @@ func GetPortsForService(detectedPorts []int32, serviceName string) []int32 {
 	}
 	return exposePorts
 }
+
+// GetPortForService returns port used by a service
+func GetPortForService(detectedPorts []int32, serviceName string) int32 {
+	var detectedPortsStr []string
+	var exposePortStr string
+	var exposePort int32
+	if len(detectedPorts) != 0 {
+		for _, detectedPort := range detectedPorts {
+			detectedPortsStr = append(detectedPortsStr, strconv.Itoa(int(detectedPort)))
+		}
+		allDetectedPortsStr := append(detectedPortsStr, qatypes.OtherAnswer)
+		exposePortStr = qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigPortForServiceKeySegment, fmt.Sprintf("Select port to be exposed for the service %s :", serviceName), []string{fmt.Sprintf("Select Other if you want to expose the service %s to some other port", serviceName)}, allDetectedPortsStr[0], allDetectedPortsStr)
+	}
+	if exposePortStr == qatypes.OtherAnswer || exposePortStr == "" {
```

Comment:
This portion is not required. The qa engine automatically does this if it finds other answer in the options.

Reply:
But, in line 134 `if len(detectedPorts) != 0 {` if the  detectedPorts is empty, then we need to check if `exposePortStr == ""` and then get the port using FetchStringAnswer. So, should I just remove the OtherAnswer condition and keep the other one? Or should I remove line 141-143?

Reply:
@ashokponkumar ji, I have updated the code. Please review it.